edweiss-app/components/quiz/LectureQuizComponents.tsx

```diff
@@ -0,0 +1,162 @@
+import { LectureQuizDisplay, LectureQuizResultDisplay, sendToLectureEvent } from '@/app/(app)/quiz/lectureQuizStudentView';
+import { SingleDistributionDisplay } from '@/app/(app)/quiz/temporaryQuizProfView';
+import { CollectionOf, Document } from '@/config/firebase';
+import ReactComponent from '@/constants/Component';
+import { useAuth } from '@/contexts/auth';
+import { useUser } from '@/contexts/user';
+import { useDoc, useDocs, usePrefetchedDynamicDoc } from '@/hooks/firebase/firestore';
+import LectureDisplay from '@/model/lectures/lectureDoc';
+import { LectureQuizzesAttempts, QuizzesAttempts } from '@/model/quizzes';
+import { Redirect } from 'expo-router';
+import { useCallback, useEffect, useState } from 'react';
+import TSafeArea from '../core/containers/TSafeArea';
+import TView from '../core/containers/TView';
+import TActivityIndicator from '../core/TActivityIndicator';
+import TText from '../core/TText';
+
+const LectureQuizView: ReactComponent<{ courseId: string, lectureId: string, lectureEventId: string }> = ({ courseId, lectureId, lectureEventId }) => {
+	const pathToEvents = "courses/" + courseId + "/lectures/" + lectureId + "/lectureEvents"
+	const pathToAttempts = pathToEvents + "/" + lectureEventId + "/attempts";
+
+	const { user } = useUser()
+	const [quizEvent, _] = usePrefetchedDynamicDoc(CollectionOf<LectureDisplay.LectureEvent>(pathToEvents), lectureEventId as string, undefined);
+
+	return (
+		<>
+			{user.type == "professor" && <>
+				<LectureQuizProfView courseId={courseId} lectureId={lectureId} lectureEventId={lectureEventId} quizEvent={quizEvent} pathToAttempts={pathToAttempts} />
+			</>}
+			{user.type == "student" && <>
+				<LectureQuizStudentView courseId={courseId} lectureId={lectureId} lectureEventId={lectureEventId} quizEvent={quizEvent} pathToAttempts={pathToAttempts} pathToLectureEvents={pathToEvents} />
+			</>}
+		</>
+	);
+};
+
+export const LectureQuizProfView: ReactComponent<{ courseId: string, lectureId: string, lectureEventId: string, quizEvent: Document<LectureDisplay.QuizLectureEvent> | undefined, pathToAttempts: string }> = ({ courseId, lectureId, lectureEventId, quizEvent, pathToAttempts }) => {
+
+
+	const studentAttempts = useDocs(CollectionOf<LectureQuizzesAttempts.LectureQuizAttempt>(pathToAttempts));
+
+	const quiz = quizEvent?.data.quizModel
+
+	useEffect(() => {
+		if (quiz?.exercise == undefined) {
+			return;
+		}
+		// else if (quizEvent?.data.done) {
+		// 	router.back();
+		// }
+	}, [quizEvent]);
+
+	// ------------The following must be managed in the remote control-----------
```

Comment:
I know there's this little message but I might be penalized if I don't point it out : you might want to remove this commented out code.

Reply:
Yes, the comment is actually intended: it is meant to be seen by Adamm